### PR TITLE
Add gateway status panel

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR adds the gateway status panel

#### Changes

- Add apexcharts
- Add panel
- Fix general styling

##### Steps

Log into the console in staging
Go to several gateways to seen different states of the panel

##### Results

Screenshot 2024-05-27 at 16 15 50
Screenshot 2024-05-27 at 16 57 52
Screenshot 2024-05-27 at 17 02 50

##### Regressions

...

#### Notes for Reviewers

This is how the responsiveness looks currently. I think this looks too crammed, and long in height. Should we consider separating the information into several panels?
Also, for this I cannot rely on our screen breakpoints, because there is a medium screen where the panels are in a 2/2 grid, and need to have the responsiveness applied, but then right after, on an even smaller screen the panels are already in a 1/1 grid and there they don't need the responsiveness. So I am using container media queries. We should define container breakpoints, because right now I am using the width of the moment the layout breaks.

Screenshot 2024-05-28 at 10 12 56

Screenshot 2024-05-28 at 10 01 10

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/components/status/index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ import PropTypes from '@ttn-lw/lib/prop-types'
 import style from './status.styl'
 
 const m = defineMessages({
+  green: 'green',
   good: 'good',
   bad: 'bad',
```
Comment (kschiffer) [inline pkg/webui/components/status/index.js]: We should remove all of these as I don't see the purpose of these titles.

File: pkg/webui/console/containers/gateway-status-panel/index.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  dutyCycleUtilization: 'Duty cycle utilization',
+  dutyCycleUtilizationTooltip:
+    '<b>What is this?</b>{lineBreak}The utilization of this gateway per sub-band.{lineBreak}All network traffic has to be in accordance with local regulations that govern the maximum usage of radio transmissions per frequency in a given time-frame. The listing allows you to inspect how much of this allowance has been exhausted already. Once utilization is exhausted, transmissions are suspended automatically by the gateway server.',
+  uptime: '30 day uptime',
+  uptimeTooltip:
+    '<b>What is this?</b>{lineBreak}The 30 day uptime expresses the relative amount of time that the gateway has been connected to the gateway server in the last 30 days.',
+  noRoundtrip: 'This gateway doesn’t have recent downlinks and cannot display the roundtrip time.',
+  noDutyCycle:
+    'This gateway doesn’t have recent downlinks and cannot display the duty cycle utilization.',
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-status-panel/index.js]:
```suggestion
    'This gateway does not have recent downlinks and cannot display the duty cycle utilization.',
```

File: pkg/webui/console/containers/gateway-status-panel/index.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+      shortCutLinkTarget="_blank"
+      divider
+      className={style.gtwStatusPanel}
+      iconClassName={isDisconnected ? style.gtwStatusPanelIcon : undefined}
+      messageDecorators={
+        <Status
+          status={isDisconnected ? 'bad' : isFetching || isUnavailable ? 'mediocre' : 'green'}
+          pulse
+          big
+          pulseTrigger={2}
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-status-panel/index.js]:
```suggestion
          pulseTrigger={gatewayStats}
```

File: pkg/webui/console/containers/gateway-status-panel/roundtrip-times.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      ? 100
+      : minRoundTripTimeInMs > 500
+        ? 0
+        : maxRoundTripTimeInMs >= 500
+          ? (400 * 100) / barWidthInMs
+          : ((maxRoundTripTimeInMs - 100) * 100) / barWidthInMs
+  const hasRed = maxRoundTripTimeInMs >= 500
+
+  return (
+    <>
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-status-panel/roundtrip-times.js:56]: You don't need the outer wrapper.
```suggestion
```

File: pkg/webui/console/containers/gateway-status-panel/transmissions.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    <>
+      <div className={style.gtwStatusPanelTransmissions}>
+        <div className="d-flex al-center gap-cs-xxs">
+          <Icon icon={IconUplink} className="c-text-neutral-semilight" />
+          {!showUplink ? (
+            <Message content={m.noUplinks} className="fw-bold" />
+          ) : (
+            <FormattedNumber value={gatewayStats.uplink_count}>
+              {parts => (
+                <>
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-status-panel/transmissions.js]:
```suggestion
```

**Review by kschiffer — CHANGES_REQUESTED** [review]
Looking good.

Please check this:
image
The roundtrip times bar is missing here for `kerlink-ibts-compact-1`.

**Review by kschiffer — CHANGES_REQUESTED** [review]
See above.

File: pkg/webui/console/containers/gateway-status-panel/roundtrip-times.js (hunk 4 above)
Comment (ryaplots) [reply pkg/webui/console/containers/gateway-status-panel/roundtrip-times.js:56]: Are you sure? It's wrapping 3 separate divs and if I remove my IDE complains with `JSX expressions must have one parent element.`

Comment (kschiffer) [reply pkg/webui/console/containers/gateway-status-panel/roundtrip-times.js:56]: Oh yeah indeed. Need to get my eyes check, I suppose!

**Review by kschiffer — APPROVED** [review]

**Pull request merged**